src/module.c

```diff
@@ -2023,6 +2023,139 @@ int RM_PublishMessage(RedisModuleCtx *ctx, RedisModuleString *channel, RedisModu
     return receivers;
 }
 
+/* Function pointer type for subscription from modules. */
+typedef int (*RedisModuleSubscribeFunc) (RedisModuleCtx *ctx, RedisModuleString *channel,  RedisModuleString *message);
+
+/* Channel subscriber information.
+ * See for more information. */
+typedef struct RedisModuleChannelSubscriber {
+    /* The module subscribed to the event */
+    RedisModule *module;
+    /* Notification callback in the module*/
+    RedisModuleSubscribeFunc subscribe_callback;
+} RedisModuleChannelSubscriber;
+
+/* The channel to module subscribers dict. */
+static dict *channelModuleSubscribers;
+
+/* The module to channel subscribers dict. */
+static dict *moduleChannelSubscribers;
```

Comment:
This seems like it would make more sense to be just a linked list on each individual module object.